File: C++/프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<string> solution(int n, vector<int> arr1, vector<int> arr2) {
    vector<string> answer;
    string row;
    
    for(int i = 0; i < arr1.size(); i++) {
        row = "";
        for(int j = n - 1; j >= 0; j--) {
            if((arr1[i] >> j) & 1 || (arr2[i] >> j) & 1) {
                row += "#";
            } else {
                row += " ";
            }
        }
        answer.push_back(row);
    }
    
    return answer;
}

File: Python3/프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for x, y in zip(arr1, arr2):
        answer.append(''.join(["#" if int(i)|int(j) else " " for i, j in zip(bin(x)[2:].zfill(n), bin(y)[2:].zfill(n))]))
        print(bin(x))
        print(bin(y))
    return answer